ibm/mas_devops/roles/gencfg_uds/tasks/main.yml

```diff
@@ -2,43 +2,39 @@
 
 # 1. Check for required facts
 # -----------------------------------------------------------------------------
-- name: "Fail if bas_endpoint_url has not been provided"
-  when: bas_endpoint_url is not defined or bas_endpoint_url|length == 0
-  fail:
-    msg: "bas_endpoint_url property has not been set"
+- name: "Fail if uds_endpoint_url has not been provided"
+  assert:
+    that: uds_endpoint_url is not defined or uds_endpoint_url|length == 0
```

Comment:
The "assert" reverses the logic that the original "when" had, so instead the "that" section should resolve to true else it will create an error. In this line it means that
```
    that: uds_endpoint_url is not defined or uds_endpoint_url|length == 0
```
should actually be
```
    that: uds_endpoint_url is defined and uds_endpoint_url|length != 0
```
i.e. "is defined" and not "is not defined". Replace the or with an and. And length should not 0